benchmarks/src/main/kotlin/montecarlo/MonteCarloBasedChannelBenchmark.kt

```diff
@@ -0,0 +1,203 @@
+            out.flush()
+            currentConfigurationNumber++
+        }
+    }
+
+    out.close()
+}
+
+fun computeStandardDeviation(list : List<Long>) : Double {
+    var sum = 0.0
+    var standardDeviation = 0.0
+    for (num in list) {
+        sum += num
+    }
+    val mean = sum / list.size
+    for (num in list) {
+        standardDeviation += (num - mean).pow(2.0)
+    }
+    return sqrt(standardDeviation / list.size)
+}
+
+/**
+ * Estimated time of arrival
+ */
+fun eta(curIteration: Int, totalIterations: Int, startTime: Long): String {
+    if (curIteration == 0) return "ETA - NaN"
+    val time = System.currentTimeMillis() - startTime
+    var eta = (time.toDouble() / curIteration * totalIterations / 60_000).toInt() // in minutes
+    val minutes = eta % 60
+    eta /= 60
+    val hours = eta % 24
+    eta /= 24
+    val days = eta
+    return "ETA - $days days, $hours hours,  $minutes minutes"
+}
+
+fun run(threads: Int, channelType: ChannelType): Long {
+    // create workload
+    val producers = Random.nextInt(1, max(threads, 2))
+    val consumers = max(1, threads - producers)
+    val (producerWorks, consumerWorks) = createWorkLoad(producers, consumers)
+
+    // start all threads
+    val phaser = Phaser(producers + consumers + 1)
+    val startTime = startWork(channelType, phaser, producers, producerWorks, consumers, consumerWorks)
+
+    // wait until the total work is done
+    phaser.arriveAndAwaitAdvance()
+    val endTime = System.nanoTime()
+    return (endTime - startTime) / 1_000_000 // ms
+}
+
+private fun startWork(channelType: ChannelType, phaser: Phaser,
+                      producers: Int, producerWorks: List<Int>,
+                      consumers: Int, consumerWorks: List<Int>) : Long {
+    val channel = channelType.createChannel<Int>()
+    val totalMessagesCount = APPROX_BATCH_SIZE / (producers * consumers) * (producers * consumers)
+    check(totalMessagesCount / producers * producers == totalMessagesCount)
+    check(totalMessagesCount / consumers * consumers == totalMessagesCount)
+    val startTime = System.nanoTime()
+
+    // run producers
+    repeat(producers) { prodId ->
+        val work = producerWorks[prodId]
+        Thread {
+            repeat(totalMessagesCount / producers) {
+                runBlocking {
+                    channel.send(it)
+                }
+                doWork(work)
+            }
+            phaser.arrive()
+        }.start()
+    }
+
+    // run consumers
+    repeat(consumers) { consId ->
+        val work = consumerWorks[consId]
+        Thread {
+            repeat(totalMessagesCount / consumers) {
+                runBlocking {
+                    channel.receive()
+                }
+                doWork(work)
+            }
+            phaser.arrive()
+        }.start()
+    }
+
+    return startTime
+}
+
+fun createWorkLoad(producers: Int, consumers: Int): Pair<List<Int>, List<Int>> {
```

Comment:
Do we really need to balance the system for coroutines? Please, write both pros and cons, so that we can decide.